### PR TITLE
Introduce dynamic config rollout percentage for history branch operations

### PR DESCRIPTION
**Pull request by norberthu**
**What changed?**

**Why?**
Prefactor for the following soon to be introduced history branch related RPC:
- ReadHistoryBranch
- ReadHistoryBranchReverse
- ReadRawHistoryBranch
- TrimHistoryBranch
- DeleteHistoryBranch

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: config/dynamicconfig/development-cass.yaml [hunk 1]
```diff
@@ -32,3 +32,5 @@
 #  - value: true
 system.enableEagerWorkflowStart:
   - value: true
+frontend.percentageFrontendToHistoryRPC:
+  - value: 0.0
```
[the remark this answers is not in the document]
Comment (norberthu) [reply config/dynamicconfig/development-cass.yaml:36]: I used this during development to dial the traffic back and forth. I'll remove it from the PR

File: common/dynamicconfig/constants.go [hunk 2]
```diff
@@ -277,6 +277,9 @@ const (
 	// FrontendEnableUpdateWorkflowExecution enables worker versioning data read / write APIs.
 	FrontendEnableWorkerVersioningDataAPIs = "frontend.workerVersioningDataAPIs"
 
+	// FrontendReadEventsFromHistoryFraction (0.0~1.0) frontend calls to history for history branch related operations.
```
Comment (MichaelSnowden) [inline common/dynamicconfig/constants.go:280]:
```suggestion
	// FrontendReadEventsFromHistoryFraction (0.0~1.0) is the fraction of history branch queries that are sent to the history service. Other queries hit the database directly.
```

File: service/frontend/adminHandler.go [hunk 3]
```diff
@@ -948,42 +948,52 @@ func (adh *AdminHandler) GetWorkflowExecutionRawHistoryV2(ctx context.Context, r
...
+				// when no events can be returned from DB, DB layer will return
+				// EntityNotExistsError, this API shall return empty response
+				return &adminservice.GetWorkflowExecutionRawHistoryV2Response{
+					HistoryBatches: []*commonpb.DataBlob{},
+					NextPageToken:  nil, // no further pagination
+					VersionHistory: targetVersionHistory,
+				}, nil
+			}
+			return nil, err
 		}
```
Comment (MichaelSnowden) [inline service/frontend/adminHandler.go:979]: I don't think this should be committed--what happens if someone sets the dc to 100% in a later release and then rolls back to this one? Or, what if someone sets the DC to a non-zero number now? We should add the implementation first, then the DC branching

**Review by MichaelSnowden — CHANGES_REQUESTED** [review]

**Review by MichaelSnowden — APPROVED** [review]
LGTM after linter comment

**Pull request merged**